base_external_system_oauth/models/auth_oauth_provider.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020-2024 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+# pylint: disable=no-self-use
+"""Provide extra functionality to use Microsoft oauth for applications."""
+import logging
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class AuthOAuthProvider(models.Model):
+    """Abstract model to communicate with oauth authentication provider."""
+
+    _inherit = "auth.oauth.provider"
+
+    tenant_id = fields.Char()
+    client_secret = fields.Char()
+    provider_type = fields.Selection(
+        selection=[
+            ("demo", "DEMO Just for testing purposes"),
+            ("microsoft_client_secret", "Microsoft Client Secret Authentication"),
+        ],
+        default="basic",
```

Comment:
Bug here, it should be 'demo'

Reply:
Ah, yes, I removed the useless (in this context) basic method, but forgot to adapt this one.